toolchain/check/handle_function.cpp

```diff
@@ -90,15 +91,12 @@ static auto FindSelfPattern(Context& context,
     -> SemIR::InstId {
   auto implicit_param_patterns =
       context.inst_blocks().GetOrEmpty(implicit_param_patterns_id);
-  if (const auto* i = llvm::find_if(implicit_param_patterns,
-                                    [&](auto implicit_param_id) {
-                                      return SemIR::IsSelfPattern(
-                                          context.sem_ir(), implicit_param_id);
-                                    });
-      i != implicit_param_patterns.end()) {
-    return *i;
-  }
-  return SemIR::InstId::None;
+  return FindIfOrDefault(
```

Comment:
I wonder how often we'd use a `FindIfOrNone` variant that returns `value_type::None` if not found.

Reply:
Hmm that's an interesting way to frame it. Added and used that. Dropping the OrDefault flavour of both finds for now.